distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -128,10 +195,29 @@ public Response apply(Response response) {
         }
 
         private Response respond(Response response) {
-            for (char c : responseMessage.toCharArray()) {
+            Map<String, String> requestHeaders = getRequestHeaders.apply(null);
+            LOGGER.debug("StubServer: extracted request headers [{}]", requestHeaders);
+
+            // if range header is present, return 206 - Partial Content status and set Content-Range header if byte Offset is specified
+            ByteOffset byteOffset;
+            if (requestHeaders != null && !StringUtil.isBlank(requestHeaders.get("range"))) {
+                byteOffset = new ByteOffset(requestHeaders.get("range"), responseMessage.length);
+                response.setStatus(HttpStatus.PARTIAL_CONTENT_206);
+                response.setHeader("Content-Range", byteOffset.contentRangeValue());
+                LOGGER.debug("StubServer: Response range header set to [Content-Range: {}]",
+                        byteOffset.contentRangeValue());
+            } else {
+                response.setStatus(200); //TODO: change this to a constant
+                byteOffset = new ByteOffset(0, responseMessage.length - 1);
+                LOGGER.debug("StubServer: set response status to 200");
+            }
+
+            // send each character, respecting the range header
+            for (int i = byteOffset.start; i <= byteOffset.end; i++) {
```

Comment:
Could we move this inside the `ByteOffset` class? All we would have to do then is:

```
boolean fail = (numberOfRetries < numberOfFailures);
byteOffset.send(responseMessage, messageDelay, fail)
```